modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/xml/XmlExtensionLoaderDelegate.java

```diff
@@ -398,7 +401,13 @@ private Optional<ExtensionModel> createTnsExtensionModel(URL resource, Set<Exten
       throws IOException {
     final ByteArrayOutputStream resultStream = new ByteArrayOutputStream();
     final Transformer transformer = FOR_TNS_XSTL_TRANSFORMER_POOL.take();
-    try (BufferedInputStream resourceIS = new BufferedInputStream(resource.openStream())) {
+    URLConnection urlConnection = resource.openConnection();
```

Comment:
Move this duplicated code to an utility function in org.mule.runtime.core.api.util.IoUtils